src/lib.rs

```diff
@@ -179,16 +190,21 @@
 //! implementation:
 //!
 //! ``` ignore
+//! extern crate embedded_hal as hal;
+//! extern crate nb;
+//!
 //! /// A synchronized serial interface
 //! // NOTE This is a global singleton
-//! pub struct Serial1;
+//! pub struct Serial1(Mutex<..>);
```

Comment:
If this contains a field then it's no longer a global singleton. It's also not required to have this field. See below